File: flapypan-blog-api/src/main/java/top/flapypan/blog/controller/SettingController.java
package top.flapypan.blog.controller;

import cn.dev33.satoken.annotation.SaIgnore;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;
import top.flapypan.blog.common.RestResult;
import top.flapypan.blog.service.SettingService;

import java.util.Map;

@Slf4j
@RequiredArgsConstructor
@RestController
@RequestMapping("/setting")
public class SettingController {

    private final SettingService settingService;

    @SaIgnore
    @GetMapping
    public RestResult<Map<String, String>> getSettings() {
        return RestResult.success(settingService.getSettingsMap());
    }

    @PostMapping
    public RestResult<Void> updateSettings(@RequestBody Map<String, String> settingsMap) {
        settingService.saveSettingsMap(settingsMap);
        return RestResult.success();
    }

}

File: flapypan-blog-api/src/main/kotlin/top/flapypan/blog/controller/SettingController.kt
package top.flapypan.blog.controller

import org.springframework.web.bind.annotation.*
import top.flapypan.blog.common.restOk
import top.flapypan.blog.service.SettingService

/**
 * 设置相关接口
 */
@RestController
@RequestMapping("/setting")
class SettingController(
    private val settingService: SettingService
) {

    /**
     * 获取所有设置
     */
    @GetMapping
    fun getSettings() = settingService.getSettingsMap().restOk()

    /**
     * 保存设置
     */
    @PostMapping
    fun updateSettings(@RequestBody settingsMap: Map<String, String?>) =
        settingService.saveSettingsMap(settingsMap).restOk()
}
